improver/cli/phase_mask.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2020 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Interface to significant_phase_mask."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(cube: cli.inputcube, phase: str, *, model_id_attr: str = None):
+    """
+    Make significant-phase-mask cube for the specified phase.
+
+    Args:
+        cube (iris.cube.Cube):
+            The input snow-fraction data to derive the phase mask from.
+        phase (str):
+            One of "rain", "sleet" or "snow". This is the phase mask that will be
+            returned.
+        model_id_attr (str):
+            Name of the attribute used to identify the source model for
+            blending.
+    """
```

Comment:
I'm not completely convinced that it is, but we will eventually be blending engluk rain/sleet/snow data with enukx data and the blending will need this attribute to know what to do. Strictly, we don't need it on the snow-fraction or phase-mask data, but it does correctly describe them, I think.

Reply:
We'll be blending the actual diagnostic data - but those fields should have a model ID attribute already present on the probability of precipitation field, which is where they should inherit that metadata from.

I don't think this actively hurts, but in general I'd rather keep the metadata minimal.  This attribute is explicitly not in the IMPROVER standard for level 3 products (see @bjwheltor 's latest metadata audit), and we only keep it after level 1 because the blending plugin needs some way to distinguish the inputs.  (When we use EC data we may need to change this anyway, since as I understand it they use a coordinate rather than an attribute to identify the input model.)